File: testdata/fixtures/python/jwt-hardcode.py
"""jwt-python-hardcoded-secret: literal JWT HMAC secret.

jwt.encode() with a string-literal secret is reported; a secret read
from the environment is not.
"""
import os

import jwt


def mint_hardcoded(user):
    # ruleid: jwt-python-hardcoded-secret
    return jwt.encode({"sub": user}, "hardcoded-secret", algorithm="HS256")


def mint_single_quoted(user):
    # ruleid: jwt-python-hardcoded-secret
    return jwt.encode({"sub": user}, 'another-secret', algorithm="HS256")


def mint_from_env(user):
    # ok: jwt-python-hardcoded-secret
    return jwt.encode({"sub": user}, os.environ["JWT_SECRET"], algorithm="HS256")


def mint_from_config(user, secret):
    # ok: jwt-python-hardcoded-secret
    return jwt.encode({"sub": user}, secret, algorithm="HS256")

File: testdata/fixtures/java/jwt-hardcode.java
// A hard-coded HMAC secret means anyone reading the code can forge tokens.
// The rule is textual: it looks for a parenthesized `(Algorithm name)` tied
// to an `HMAC...("literal")` call, so the flagged lines keep that exact odd
// shape; a secret drawn from configuration is fine. Findings land on the
// assignment line.
import com.auth0.jwt.JWT;
import com.auth0.jwt.algorithms.Algorithm;

public class JwtHardcode {
    public String flag() {
        // ruleid: java-jwt-hardcoded-secret
        (Algorithm alg) = Algorithm.HMAC256("secret");
        return JWT.create().sign(alg);
    }

    public String flag512() {
        // ruleid: java-jwt-hardcoded-secret
        (Algorithm alg) = Algorithm.HMAC512("topsecret");
        return JWT.create().sign(alg);
    }

    public String flag384() {
        // ruleid: java-jwt-hardcoded-secret
        (Algorithm alg) = Algorithm.HMAC384("anothersecret");
        return JWT.create().sign(alg);
    }

    public String safeConfig(String secret) {
        // ok: java-jwt-hardcoded-secret
        Algorithm alg = Algorithm.HMAC256(secret);
        return JWT.create().sign(alg);
    }

    public String safeEnv() {
        // ok: java-jwt-hardcoded-secret
        Algorithm alg = Algorithm.HMAC256(System.getenv("JWT_SECRET"));
        return JWT.create().sign(alg);
    }

    public String safeRsa(java.security.interfaces.RSAPublicKey key) {
        // ok: java-jwt-hardcoded-secret
        Algorithm alg = Algorithm.RSA256(key);
        return JWT.create().sign(alg);
    }
}
